src/cli/shell.rs

```diff
@@ -0,0 +1,219 @@
+use clap::Parser;
+use directories::UserDirs;
+use std::fs::{Permissions, exists, remove_file, set_permissions};
+use std::io::Write;
+use std::os::unix::fs::PermissionsExt;
+use std::path::PathBuf;
+use std::{
+    fmt::{self},
+    fs::{self, File},
+    path::Path,
+    str::FromStr,
+};
+
+use crate::cli::common::get_default_araki_bin_dir;
+
+#[derive(Parser, Debug)]
+pub struct Args {
+    #[clap(subcommand)]
+    subcommand: ShellSubcommand,
+}
+
+#[derive(Parser, Debug, Default)]
+pub struct ShellArg {
+    shell: Option<String>,
+}
+
+#[derive(Parser, Debug)]
+pub enum ShellSubcommand {
+    // Initialize the shell configuration (by editing ~/.bashrc etc)
+    Init(ShellArg),
+
+    // Generate the environment changes to be evaluated by the shell so that the araki shims
+    // take precedence over other system binaries
+    Generate(ShellArg),
+}
+
+enum Shell {
+    Bash,
+    Zsh,
+    Unknown(String),
+}
+
+#[derive(Debug)]
+struct ParseShellError;
+
+impl FromStr for Shell {
+    type Err = ParseShellError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let result = match s.to_lowercase().as_str() {
+            "bash" => Shell::Bash,
+            "zsh" => Shell::Zsh,
+            shell => Shell::Unknown(shell.to_string()),
+        };
+
+        Ok(result)
+    }
+}
+
+impl fmt::Display for Shell {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let shell_name = match self {
+            Self::Bash => "bash",
+            Self::Zsh => "zsh",
+            Self::Unknown(name) => name,
+        };
+        write!(f, "{}", shell_name)
+    }
+}
+
+impl Shell {
+    /// Update the bash/zsh config file to fiddle the PATH so that araki shims are executed instead
+    /// of `pip`, `uv`, `pixi`, `conda`, etc.
+    ///
+    /// * `path`: Path to the config file to edit
+    fn update_posix_config(&self, path: &Path) -> Result<(), String> {
+        let contents = fs::read_to_string(path)
+            .map_err(|_| "Could not open {path} to read existing shell config.")?;
+
+        let araki_posix_config = format!(
+            "# Araki configuration\neval $(araki shell generate {})\n",
+            self
+        );
+        if !contents.contains(&araki_posix_config) {
+            let mut rcfile = File::options()
+                .append(true)
+                .open(path)
+                .map_err(|_| "Could not open {path} to modify shell config.")?;
+
+            write!(&mut rcfile, "{}", &araki_posix_config)
+                .map_err(|_| "Unable to write araki shell config to {path}")?;
+        }
+
+        let dir = get_default_araki_bin_dir()?;
+        for tool in ["pip", "uv", "pixi", "conda"] {
```

Comment:
This is a bit tricky. pip and uv are executables, so this mostly works. But this won't really work for pixi or conda.

When a user is in a terminal session with an environment activated (they have run `eval "$(araki activate thenev)"` pixi is registered as a shell function.

Conda is also always registered as a shell function. So, this shim is not able to stop the conda command from running.